samples/extensions/sparse_image/sparse_image.cpp

```diff
@@ -0,0 +1,1430 @@
+	submit_info.commandBufferCount = 1U;
+	submit_info.pCommandBuffers    = &draw_cmd_buffers[current_buffer];
+	VK_CHECK(vkQueueSubmit(queue, 1U, &submit_info, VK_NULL_HANDLE));
+	ApiVulkanSample::submit_frame();
+}
+
+std::unique_ptr<vkb::VulkanSample> create_sparse_image()
+{
+	return std::make_unique<SparseImage>();
+}
+
+/**
+ * 	@brief Generate the mesh and calculate required mip level for each texture block
+ */
+void SparseImage::calculate_mips_table(glm::mat4 mvp_transform, uint32_t numVerticalBlocks, uint32_t numHorizontalBlocks, std::vector<std::vector<MipBlock>> &mipTable)
+{
+	CalculateMipLevelData meshData(mvp_transform, VkExtent2D({static_cast<uint32_t>(virtual_texture.width), static_cast<uint32_t>(virtual_texture.height)}), VkExtent2D({static_cast<uint32_t>(width), static_cast<uint32_t>(height)}), on_screen_num_vertical_blocks, on_screen_num_horizontal_blocks, virtual_texture.mip_levels);
+
+	meshData.calculate_mesh_coordinates();
+	meshData.calculate_mip_levels();
+
+	mipTable = meshData.mip_table;
+}
+
+SparseImage::CalculateMipLevelData::CalculateMipLevelData(glm::mat4 mvp_transform, VkExtent2D texture_base_dim, VkExtent2D screen_base_dim, uint32_t vertical_num_blocks, uint32_t horizontal_num_blocks, uint8_t mip_levels) :
+    mvp_transform(mvp_transform), texture_base_dim(texture_base_dim), screen_base_dim(screen_base_dim), mesh{0}, vertical_num_blocks(vertical_num_blocks), horizontal_num_blocks(horizontal_num_blocks), mip_levels(mip_levels)
+{
+	mesh.resize(vertical_num_blocks + 1U);
+	for (auto &row : mesh)
+	{
+		row.resize(horizontal_num_blocks + 1U);
+	}
+
+	ax_vertical.resize(horizontal_num_blocks + 1U);
+	ax_horizontal.resize(vertical_num_blocks + 1U);
+}
+
+void SparseImage::CalculateMipLevelData::calculate_mesh_coordinates()
+{
+	glm::vec4 top_left(-100.0f, -100.0f, 0.0f, 1.0f);
+	glm::vec4 top_right(100.0f, -100.0f, 0.0f, 1.0f);
+	glm::vec4 bottom_left(-100.0f, 100.0f, 0.0f, 1.0f);
+	glm::vec4 bottom_right(100.0f, 100.0f, 0.0f, 1.0f);
+
+	float h_interval = (top_right[0] - top_left[0]) / horizontal_num_blocks;
+	float v_interval = (bottom_left[1] - top_left[1]) / vertical_num_blocks;
+
+	for (size_t v_index = 0U; v_index < vertical_num_blocks + 1U; v_index++)
+	{
+		for (size_t h_index = 0U; h_index < horizontal_num_blocks + 1U; h_index++)
+		{
+			float x_norm = top_left[0] + h_index * h_interval;
+			float y_norm = top_left[1] + v_index * v_interval;
+
+			glm::vec4 result = mvp_transform * glm::vec4(x_norm, y_norm, 0.0f, 1.0f);
+
+			double x = 0.5 * screen_base_dim.width * result[0] / abs(result[3]);
+			double y = 0.5 * screen_base_dim.height * result[1] / abs(result[3]);
+
+			mesh[v_index][h_index].x = x;
+			mesh[v_index][h_index].y = y;
+
+			mesh[v_index][h_index].on_screen = (x < (screen_base_dim.width / (-2.0))) || (x > (screen_base_dim.width / 2.0)) || (y < (screen_base_dim.height / (-2.0))) || (y > (screen_base_dim.height / 2.0)) || (result[3] < 0.0) ? false : true;
+		}
```

Comment:
Maybe this is a bit easier to read:
`			mesh[v_index][h_index].on_screen = (-half_width < x) && (x < half_width) && (-half_height < y) && (y < half_height) && (0.0 < result[3]);
`
With some `half_width` and `half_height`, calculated in front of the double loop.